supervision/metrics/mean_average_recall.py

```diff
@@ -0,0 +1,452 @@
+
+        Returns:
+            (MeanAverageRecall): The updated metric instance.
+        """
+        if not isinstance(predictions, list):
+            predictions = [predictions]
+        if not isinstance(targets, list):
+            targets = [targets]
+
+        if len(predictions) != len(targets):
+            raise ValueError(
+                f"The number of predictions ({len(predictions)}) and"
+                f" targets ({len(targets)}) must be the same."
+            )
+
+        if self._class_agnostic:
+            predictions = deepcopy(predictions)
+            targets = deepcopy(targets)
+            for prediction in predictions:
+                prediction.class_id[:] = -1
+            for target in targets:
+                target.class_id[:] = -1
+
+        self._predictions_list.extend(predictions)
+        self._targets_list.extend(targets)
+        return self
+
+    def compute(self) -> MeanAverageRecallResult:
+        """
+        Calculate Mean Average Recall based on predicted and ground-truth
+        detections at different thresholds.
+
+        Returns:
+            (MeanAverageRecallResult): The Mean Average Recall result.
+        """
+        result = self._compute(self._predictions_list, self._targets_list)
+
+        for size_category, container in [
+            (ObjectSizeCategory.SMALL, "small_objects"),
+            (ObjectSizeCategory.MEDIUM, "medium_objects"),
+            (ObjectSizeCategory.LARGE, "large_objects"),
+        ]:
+            size_predictions = []
+            size_targets = []
+            for predictions, targets in zip(self._predictions_list, self._targets_list):
+                size_predictions.append(
+                    self._filter_detections_by_size(predictions, size_category)
+                )
+                size_targets.append(
+                    self._filter_detections_by_size(targets, size_category)
+                )
+            setattr(result, container, self._compute(size_predictions, size_targets))
+
+        return result
+
+    def _compute(
+        self,
+        predictions_list: List[Detections],
+        targets_list: List[Detections],
+    ) -> MeanAverageRecallResult:
+        if not predictions_list or not targets_list:
+            return MeanAverageRecallResult(
+                metric_target=self._metric_target,
+                is_class_agnostic=self._class_agnostic,
+                mean_average_recall=0.0,
+                ar_per_class=np.array([]),
+                matched_classes=np.array([]),
+            )
+
+        all_recalls = []
+        all_class_ids = []
+        iou_thresholds = np.linspace(0.5, 0.95, 10)
+
+        for predictions, targets in zip(predictions_list, targets_list):
+            if targets.is_empty():
+                continue
+
+            prediction_contents = self._detections_content(predictions)
+            target_contents = self._detections_content(targets)
```

Comment:
Idea for later: in all metrics, we can move these 2 lines lower by a block (to  `if self._metric_target == MetricTarget.BOXES:`), and then extract into a function `_compute_iou`.

(No changes needed now.)